# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFont.cpp
# Commit: 793e9dfa4b9f
# Full Hash: 793e9dfa4b9fa40137cdf2a5d34c126d6cc9840d
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-02-29 04:04:56
# Regressor Bug: 1616995
# File Overlap Count: 1
# Description:
#   Bug 1616995 - patch 2 - Support vertical skew for upright-vertical fonts in webrender. r=lsalzman
#   
#   Depends on D63893
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D64746
# ==============================================================================

diff -r 88d3197f16c2 -r 793e9dfa4b9f gfx/thebes/gfxFont.cpp
--- a/gfx/thebes/gfxFont.cpp	Fri Feb 28 14:55:39 2020 +0000
+++ b/gfx/thebes/gfxFont.cpp	Fri Feb 28 15:36:19 2020 +0000
@@ -2182,6 +2182,11 @@
       float baseAdj = (metrics.emAscent - metrics.emDescent) / 2;
       baseline += baseAdj * aTextRun->GetAppUnitsPerDevUnit() * baselineDir;
     }
+  } else if (textDrawer &&
+             aOrientation == ShapedTextFlags::TEXT_ORIENT_VERTICAL_UPRIGHT) {
+    glyphFlagsRestore.Save(textDrawer);
+    textDrawer->SetWRGlyphFlags(textDrawer->GetWRGlyphFlags() |
+                                wr::FontInstanceFlags::VERTICAL);
   }
 
   if (fontParams.obliqueSkew != 0.0f && !fontParams.isVerticalFont &&